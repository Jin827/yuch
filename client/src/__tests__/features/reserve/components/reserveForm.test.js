import React from 'react';
import { shallow } from 'enzyme';
import { unwrap } from '@material-ui/core/test-utils';
import ReserveForm from '../../../../features/reserve/components/reserveForm';
import * as mockData from '../../../__mocks__/mockData';

describe('<ReserveForm />', () => {
  const mockClose = jest.fn();
  const mockChange = jest.fn();
  const mockSubmit = jest.fn();

  const ComponentNaked = unwrap(ReserveForm);

  const props = {
    reserveInfo: mockData.reserveInfoInit,
    inThreeDays: mockData.inThreeDays,
    handleClose: mockClose,
    handleChange: mockChange,
    handleSubmit: mockSubmit,
    classes: {},
  };
  const wrapper = shallow(<ComponentNaked {...props} />);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('creates inputs', () => {
    expect(wrapper.find('.input-name').exists()).toBe(true);
    expect(wrapper.find('.input-contact').exists()).toBe(true);
    expect(wrapper.find('.input-number').exists()).toBe(true);
    expect(wrapper.find('.input-place').exists()).toBe(true);
    expect(wrapper.find('.input-date').exists()).toBe(true);
    expect(wrapper.find('.input-time').exists()).toBe(true);
  });

  describe('when typeing into input, calls a callback function', () => {
    it('name', () => {
      const ev = { target: { id: 'name', value: 'Yuchung' } };
      wrapper.find('.input-name').simulate('change', ev);
      expect(mockChange).toHaveBeenCalledWith(ev);
    });

    it('contact', () => {
      const ev = { target: { id: 'contact', value: '123123123' } };
      wrapper.find('.input-contact').simulate('change', ev);
      expect(mockChange).toHaveBeenCalledWith(ev);
    });

    it('number', () => {
      const ev = { target: { id: 'number', value: '100' } };
      wrapper.find('.input-number').simulate('change', ev);
      expect(mockChange).toHaveBeenCalledWith(ev);
    });

    it('place', () => {
      const ev = { target: { id: 'place', value: 'Gyeongju' } };
      wrapper.find('.input-place').simulate('change', ev);
      expect(mockChange).toHaveBeenCalledWith(ev);
    });

    it('date', () => {
      const ev = { target: { id: 'date', value: '2019-12-25' } };
      wrapper.find('.input-date').simulate('change', ev);
      expect(mockChange).toHaveBeenCalledWith(ev);
    });

    it('time', () => {
      const ev = { target: { id: 'time', value: '2:30 PM' } };
      wrapper.find('.input-time').simulate('change', ev);
      expect(mockChange).toHaveBeenCalledWith(ev);
    });
  });

  describe('when clicking button, calls a callback function', () => {
    const mockedEvent = { target: {} };

    it('reserve button', () => {
      wrapper.find('.btn-reserve').simulate('click', mockedEvent);
      expect(mockSubmit).toHaveBeenCalledWith(mockedEvent);
    });

    it('close button', () => {
      wrapper.find('.btn-close').simulate('click');
      expect(mockClose).toHaveBeenCalledWith();
    });
  });
});
