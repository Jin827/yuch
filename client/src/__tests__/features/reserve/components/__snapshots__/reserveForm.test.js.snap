// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReserveForm /> renders correctly 1`] = `
<Fragment>
  <div>
    <TextField
      InputLabelProps={
        Object {
          "shrink": true,
        }
      }
      className="input-name"
      fullWidth={true}
      id="name"
      label="이름"
      margin="normal"
      onChange={[Function]}
      required={true}
      select={false}
      value=""
      variant="standard"
    />
    <WithStyles(FormControl)
      fullWidth={true}
      margin="normal"
    >
      <WithStyles(InputLabel)
        htmlFor="formatted-text-mask-input"
      >
        연락처 *
      </WithStyles(InputLabel)>
      <WithStyles(Input)
        className="input-contact"
        id="contact"
        inputComponent={[Function]}
        onChange={[Function]}
        placeholder="(054)-745-0999"
        required={true}
        value="(0  )    -    "
      />
    </WithStyles(FormControl)>
    <TextField
      InputLabelProps={
        Object {
          "shrink": true,
        }
      }
      className="input-number"
      fullWidth={true}
      id="number"
      label="인원수"
      margin="normal"
      onChange={[Function]}
      required={true}
      select={false}
      type="number"
      value=""
      variant="standard"
    />
    <TextField
      InputLabelProps={
        Object {
          "shrink": true,
        }
      }
      className="input-place"
      fullWidth={true}
      id="place"
      label="장소"
      margin="normal"
      multiline={true}
      onChange={[Function]}
      required={true}
      select={false}
      value=""
      variant="standard"
    />
    <TextField
      className="input-date"
      fullWidth={true}
      id="date"
      label="날짜"
      margin="normal"
      onChange={[Function]}
      required={true}
      select={false}
      type="date"
      variant="standard"
    />
    <TextField
      className="input-time"
      defaultValue="12:30"
      fullWidth={true}
      id="time"
      inputProps={
        Object {
          "step": 900,
        }
      }
      label="시간"
      margin="normal"
      onChange={[Function]}
      required={true}
      select={false}
      type="time"
      variant="standard"
    />
  </div>
  <div
    className="mt2"
  >
    <WithStyles(Button)
      className="btn-reserve undefined"
      color="secondary"
      onClick={[Function]}
      variant="contained"
    >
      예약완료
    </WithStyles(Button)>
    <WithStyles(Button)
      className="btn-close undefined"
      color="secondary"
      onClick={[MockFunction]}
      variant="outlined"
    >
      닫기
    </WithStyles(Button)>
  </div>
</Fragment>
`;
